operator/v1/types_csi_cluster_driver.go

```diff
@@ -96,6 +96,55 @@ type ClusterCSIDriverSpec struct {
 	// The current default behaviour is Managed.
 	// +optional
 	StorageClassState StorageClassStateName `json:"storageClassState,omitempty"`
+
+	// driverConfig can be used to specify platform specific driver configuration.
+	// When omitted, this means no opinion and the platform is left to choose reasonable
+	// defaults. These defaults are subject to change over time.
+	// +optional
+	DriverConfig CSIDriverConfigSpec `json:"driverConfig,omitempty"`
+}
+
+// CSIDriverType indicates type of CSI driver being configured.
+// +kubebuilder:validation:Enum=vsphere
+type CSIDriverType string
+
+const (
+	VSphereDriverType CSIDriverType = "vsphere"
+)
+
+// CSIDriverConfigSpec defines configuration spec that can be
+// used to optionally configure a specific CSI Driver.
+// +union
+type CSIDriverConfigSpec struct {
+	// driverType indicates type of CSI driver for which the
+	// driverConfig is being applied to.
+	//
+	// Valid values are:
+	//
+	// * vsphere
+	//
+	// Allows configuration of vsphere CSI driver topology.
+	//
+	// +kubebuilder:validation:Required
+	// +unionDiscriminator
+	DriverType CSIDriverType `json:"driverType"`
```

Comment:
As this enum is expected over time, you need to add a developer note to explain what consumers should do when they observe an unknown value, there's typically two options there
```suggestion
	// ---
	// + Consumers should treat unknown values as a no-op || Consumers should treat unknown values as an error and stop processing.
	DriverType CSIDriverType `json:"driverType"`
```